barretenberg/cpp/src/barretenberg/common/thread_utils.hpp

```diff
@@ -0,0 +1,15 @@
+#include "thread.hpp"
+
+const size_t MIN_ITERS_PER_THREAD = 1 << 4;
```

Comment:
The appropriate value for MIN_ITERS_PER_THREAD may depend heavily on what computation is performed in a single 'iteration'. You could let this be the default value for the min_iterations_per_thread input variable but it's probably best to define a unique value depending on the context. (Also, I'd be surprised if it was worth spinning up a new thread for 16 field additions but we'd have to do some testing to know for sure)

Reply:
Tested a little bit with 1, 16, and 64. It seemed like 1 and 16 performed similarly and 64 had some worse results, but I only tested ~6 repetitions and it might be from server noise.